### ISSUE TITLE
should we have a predefined method for setting status to ok after remediation?

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/preservation_catalog#443]
in the discussion w/ @ndushay and @julianmorley that spawned #431 and #438, i'd suggested the idea of a canned method to set the status of a catalog entry (`PreservedCopy`) back to `OK_STATUS`.  maybe in `PreservedObjectHandler`?

two ideas:
* somewhat more involved
  * POH has class method.  takes druid and endpoint, gets current version and size on disk(`Stanford::StorageServices.current_version(druid)`, `Stanford::StorageServices.object_size(druid)`).
  * uses provided endpoint, detected current version, and detected size to create a new POH instance.
  * calls new instance method `POH#validate_and_mark_ok`
  * `#validate_and_mark_ok`
    * validates to the degree it can (including checksums once we can do those?)
    * if validation passes, in a db transaction
      * sets the version and size fields on the pres copy
      * sets the pres copy status to ok
      * don't do the on disk validation in the transaction because it's not like our reads to disk will be transactional anyway, and it'll just block the db operation unnecessarily.
    * log success or failure for the status remediation attempt based on validation and transaction success (this seemed to be the main thing @julianmorley wanted to call out in logging when status changes once things go good)
  * requires that the caller has already properly remediated the object (e.g. pulled a good copy from another endpoint to replace the bad copy, sworn an oath that they've remediated the object to the best of their knowledge, so help them god)
  * maybe an "in remediation" status on a pres copy catalog entry?  to warn both machines and other humans M2C/C2M away from bothering with the record (or object on disk)?
    * could have similar/simpler method for marking the "in remediation" status

* alternate idea: simple public POH instance method for updating status that basically calls private `update_status`, `update_db_object`, and does result logging/return
  * upside: simpler, extends/replicates easily to support an additional "in remediation" status if that was desired from the above set of ideas but the rest was too much
  * downside: if manual remediation of pres copy fields is necessary, this makes that even more manual, and misses the opportunity for easier safeguards and logging (e.g., the record was always bad, and then size and/or current version have to be updated manually with random active record statements).

**Comment by ndushay** [comment]
See also #433

**Comment by ndushay** [comment]
Is this really part of recovery / remediation implementation and thus postponable?

**Comment by jmartin-sul** [comment]
i believe it is.  though i'd hope that it's postponable in the sense that it still falls within this preservation work effort, even if it isn't done immediately as part of M2C.  as opposed to entirely postponable, and not part of this preservation work effort.  my thinking being: this seems like *way* less effort than full-blown automatic remediation/recovery, but it seems like it'd add a decent amount of safeguarding to the manual remediation process.  or, maybe this is yet another can of worms, and we'll regret opening it at all.  but i'm leery of remediators setting individual fields in the database manually, esp if common use cases could be packaged up into something that's tested and documented.

**Comment by ndushay** [comment]
sounds like a question for storytime.

**Comment by jmartin-sul** [comment]
this was filed under the idea that moab-to-catalog checks would just skip over anything in a bad status.  we ended up doing sort of the opposite of that idea in #456, with some simple automatic status detection and remediation.

after some discussion with @LynnMcRae and some looking at the current code, we determined:
* a lot of what's described above is actually implemented by the `set_status_as_seen_on_disk` method from PR #456.
* lynn says people shouldn't just blindly set status back to OK: the checks should be run and the code should ensure that things actually are ok (as described in the more involved option above, or as largely happens in `set_status_as_seen_on_disk`).  this at least rules out the simpler option described above.
  * but personally i do think it'd be nice for the remediator to ask for the system to detect that the object has gone OK, manually, as part of the manual object remediation, instead of just letting the system detect it on a future automatic iteration of M2C or C2M.
  * but also  lynn thinks there might also be some object re-accessioning when remediating bad moabs, which might trigger some of the other automated processes that detect status anyway?
* we might expose this functionality via REST so that e.g. argo could hit a URL and ask that status be re-set to whatever it actually is on disk at the moment.

for now, we're closing this issue.  we can file a new issue and link to this when we have specific follow-on work on deck.
